python/src/aac/plugins/rest_api/aac_rest_app.py

```diff
@@ -1,22 +1,113 @@
+from aac.io.files.find import find_aac_files, is_aac_file
+from aac.io.paths import sanitize_filesystem_path
+from aac.io.parser import parse
+from aac.io.writer import write_definitions_to_file
 from aac.lang.active_context_lifecycle_manager import get_active_context
-from aac.lang.definitions.definition import Definition
-from aac.plugins.rest_api.definition_model import DefinitionModel, to_definition_class, to_definition_model
+from aac.plugins.rest_api.models.definition_model import DefinitionModel, to_definition_class, to_definition_model
+from aac.plugins.rest_api.models.file_model import FileModel, FilePathModel, FilePathRenameModel, to_file_model
 
 app = FastAPI()
 
+AVAILABLE_AAC_FILES: list[AaCFile] = []
+WORKSPACE_DIR: str = os.getcwd()
+
+
+# File CRUD Operations
 
-@app.get("/files", status_code=HTTPStatus.OK)
-def get_files():
+@app.get("/files/context", status_code=HTTPStatus.OK, response_model=list[FileModel])
+def get_files_from_context():
     """Return a list of all files contributing definitions to the active context."""
-    file_names = [definition.source_uri for definition in get_active_context().definitions]
-    return {"files": set(file_names)}
+    return [to_file_model(file) for file in get_active_context().get_files_in_context()]
+
+
+@app.get("/files/available", status_code=HTTPStatus.OK, response_model=list[FileModel])
+def get_available_files(background_tasks: BackgroundTasks):
+    """Return a list of all files available in the workspace for import into the active context. The list of files returned does not include files already in the context."""
+    # Update the files via an async function so that any changes to the files shows up, eventually.
+    background_tasks.add_task(refresh_available_files_in_workspace)
+
+    #  Having to use a cached response for now as the file-walking makes the response take too long.
+    return [to_file_model(file) for file in AVAILABLE_AAC_FILES]
+
+
+@app.get("/file", status_code=HTTPStatus.OK, response_model=FileModel)
+def get_file_by_uri(uri: str):
+    """Return the target file from the workspace, or HTTPStatus.NOT_FOUND if the file isn't in the context."""
+    file_in_context = get_active_context().get_file_in_context_by_uri(uri)
+
+    if file_in_context:
+        file_model = to_file_model(file_in_context)
+        with open(file_in_context.uri) as file:
+            file_model.content = file.read()
+
+        return file_model
+    else:
+        _report_error_response(HTTPStatus.NOT_FOUND, f"File {uri} not found in the context.")
+
+
+@app.post("/files/import", status_code=HTTPStatus.NO_CONTENT)
+def import_files_to_context(file_models: list[FilePathModel]):
+    """Import the list of files into the context."""
+    files_to_import = set([str(model.uri) for model in file_models])
+    valid_aac_files = set(filter(is_aac_file, files_to_import))
+    invalid_files = files_to_import.difference(valid_aac_files)
+
+    if len(invalid_files) > 0:
```

Comment:
I think we should import valid files, even if there are invalid ones.

Reply:
Committing partial state alterations based from an invalid request is not a good pattern; it'll complicate error handling and batching requests for clients.

Reply:
Fair enough. I was thinking that since (I'm pretty sure) adding definitions, etc is idempotent it wouldn't be a big deal but I guess there's a good argument for atomic operations, too.